src/SonarScanner.MSBuild.Shim/PropertiesFileGenerator.cs

```diff
@@ -420,32 +419,38 @@ public DirectoryInfo ComputeRootProjectBaseDir(IEnumerable<DirectoryInfo> projec
         /// Loads SARIF reports from the given projects and attempts to fix
         /// improper escaping from Roslyn V1 (VS 2015 RTM) where appropriate.
         /// </summary>
-        private void TryFixSarifReport(ProjectInfo project, string language, string reportFilePropertyKey)
+        private void TryFixSarifReport(ProjectInfo project, string language, string reportFilesPropertyKey)
         {
-            var tryResult = project.TryGetAnalysisSetting(reportFilePropertyKey, out Property reportPathProperty);
+            var tryResult = project.TryGetAnalysisSetting(reportFilesPropertyKey, out Property reportPathsProperty);
             if (tryResult)
             {
-                foreach (var reportPath in reportPathProperty.Value.Split(RoslynReportPathsDelimiter))
+                var newPropertyValueBuilder = new StringBuilder();
+                project.AnalysisSettings.Remove(reportPathsProperty);
+                foreach (var reportPath in reportPathsProperty.Value.Split(RoslynReportPathsDelimiter))
                 {
                     var fixedPath = fixer.LoadAndFixFile(reportPath, language);
 
-                    if (!reportPath.Equals(fixedPath)) // only need to alter the property if there was no change
+                    if (!reportPath.Equals(fixedPath)) // only need to alter the property if there was a change
                     {
-                        // remove the property ahead of changing it
-                        // if the new path is null, the file was unfixable and we should leave the property out
-                        project.AnalysisSettings.Remove(reportPathProperty);
-
                         if (fixedPath != null)
                         {
-                            // otherwise, set the property value (results in no change if the file was already valid)
-                            var newReportPathProperty = new Property
-                            {
-                                Id = reportFilePropertyKey,
-                                Value = fixedPath,
-                            };
-                            project.AnalysisSettings.Add(newReportPathProperty);
+                            newPropertyValueBuilder.Append($"{fixedPath}{RoslynReportPathsDelimiter}");
                         }
                     }
+                    else
+                    {
+                        newPropertyValueBuilder.Append($"{reportPath}{RoslynReportPathsDelimiter}");
+                    }
```

Comment:
This entire nested `if/if/else` section is obfuscated/hard to read. If I read it correctly, it can be actually reduced to this:
```C#
if (fixedPath != null)
{
    newPropertyValueBuilder.Append($"{fixedPath}{RoslynReportPathsDelimiter}");
}
```

Reply:
The trick is, that `reportPath` is `fixedPath` in the `else`, so both `Append` are doing exactly the same thing.

Reply:
We just don't want to put anything in case the fix file stuff has returned null. I don't see other way to achieve that.

Reply:
This is the way to achieve that. Remove if/if/else and replace with:
```
if (fixedPath != null)
{
    newPropertyValueBuilder.Append($"{fixedPath}{RoslynReportPathsDelimiter}");
}
```

Reply:
And what is report path is equal to the "fixedpath" ? We still want to add this original value as well.

Reply:
We had a call to align about this and it's fixed now